### PR TITLE
SENT-75 [FE/BE] Implement Saved Posts

### PR DESCRIPTION
**Pull request by justinnas**
Implemented post saving functionality, saved posts page. Fixed issues with liking posts.

### CODE REVIEW

File: back-end/src/routes/saveRoute.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const saveRouter = Router();
+
+saveRouter.post('/', checkOwnership('saves'), createSave);
+
+saveRouter.get('/', checkOwnership('saves'), readSave);
+
+saveRouter.get('/savedBy', readSavedBy);
+
+saveRouter.get('/userSaves', readUserSaves);
```
Comment (mantvydasdeltuva) [inline back-end/src/routes/saveRoute.ts]: Shouldn't it check for ownership?

File: back-end/src/routes/saveRoute.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { checkOwnership } from '../middlewares/authentication';
+import { createSave, deleteSave, readSave, readSavedBy, readUserSaves } from '../APIs/controllers';
+
+export const saveRouter = Router();
+
+saveRouter.post('/', checkOwnership('saves'), createSave);
+
+saveRouter.get('/', checkOwnership('saves'), readSave);
+
+saveRouter.get('/savedBy', readSavedBy);
```
Comment (mantvydasdeltuva) [inline back-end/src/routes/saveRoute.ts]: Why is this needed?

File: back-end/src/routes/postRoute.ts [hunk 3]
```diff
@@ -6,13 +6,16 @@ import {
...
 	readPostsFollowing,
+	readPostsSaved,
 } from '../APIs/controllers/postController';
 import { checkOwnership } from '../middlewares/authentication';
 
 export const postRouter = Router();
 
 postRouter.get('/following/', readPostsFollowing);
 
+postRouter.get('/saved/', readPostsSaved);
```
Comment (mantvydasdeltuva) [inline back-end/src/routes/postRoute.ts:17]: Shouldn't it check for ownership?

File: back-end/src/APIs/controllers/postController.ts [hunk 4]
```diff
@@ -95,6 +96,37 @@ export const readPostsFollowing = async (req: Request, res: Response) => {
 	}
 };
 
+export const readPostsSaved = async (req: Request, res: Response) => {
+	try {
+		const authPayload = req.auth?.payload;
```
Comment (mantvydasdeltuva) [inline back-end/src/APIs/controllers/postController.ts:101]: Check this in authorization

File: back-end/src/database/migrations/20240412190405__VERSION_06.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import type { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+	await knex.schema.dropTableIfExists('saves');
+
+	await knex.schema.createTable('saves', (table) => {
```
Comment (mantvydasdeltuva) [inline back-end/src/database/migrations/20240412190405__VERSION_06.ts:6]: Would be nice to populate the database so that there would be mock-data in presentation

**Review by mantvydasdeltuva — COMMENTED** [review]
Provided some insights

**Review by mantvydasdeltuva — APPROVED** [review]
Approved!

File: back-end/src/routes/saveRoute.ts (hunk 2 above)
Comment (justinnas) [reply back-end/src/routes/saveRoute.ts]: At the moment it is not needed, I left it for future functionality. But you're right, I'll remove it to reduce confusion.

File: back-end/src/routes/saveRoute.ts (hunk 1 above)
Comment (justinnas) [reply back-end/src/routes/saveRoute.ts]: You're right, great catch
